### PR TITLE
feat: consent app

### PR DESCRIPTION
**Pull request by siddhart1o1**
consent application built on top of this PR :  https://github.com/GaloyMoney/galoy/pull/2707

### CODE REVIEW

File: apps/consent/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+[ORY Hydra 5 Minute Tutorial](https://www.ory.sh/docs/hydra/5min-tutorial) to
+see how this works.
+
+## FAQ
+
+### TLS Termination
+
+You can mock TLS Termination by setting environment variable
+`MOCK_TLS_TERMINATION` to any value, for example `MOCK_TLS_TERMINATION=y`. This
+will add `X-Forwarded-Proto: https` to each HTTP Request Header.
```
Comment (nicolasburtey) [inline apps/consent/README.md:44]: probably legacy readme

File: apps/consent/app/api/csrf-invalid/route.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import type { NextApiRequest, NextApiResponse } from "next";
```
Comment (nicolasburtey) [inline apps/consent/app/api/csrf-invalid/route.ts:1]: I know it was in the hydra consent code but I'm wondering if this is something we should be keeping. feels like test or unused code

File: apps/consent/app/components/card/index.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Card.tsx
```
Comment (nicolasburtey) [inline apps/consent/app/components/card/index.tsx:1]: no need for filename here

File: apps/consent/app/components/input-component.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// InputComponent.tsx
```
Comment (nicolasburtey) [inline apps/consent/app/components/input-component.tsx:1]: no need for filename here

File: apps/consent/app/components/logo/index.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react"
+import Image from "next/image"
+
+const Logo: React.FC = () => {
+  return (
+    <div className="flex justify-center mb-4">
+      {/* TODO use Blink logo ?  */}
```
Comment (nicolasburtey) [inline apps/consent/app/components/logo/index.tsx:7]: yup

File: apps/consent/app/components/main-container/index.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// MainContent.tsx
```
Comment (nicolasburtey) [inline apps/consent/app/components/main-container/index.tsx:1]: filename

File: apps/consent/app/api/csrf-invalid/route.ts (hunk 2 above)
Comment (siddhart1o1) [reply apps/consent/app/api/csrf-invalid/route.ts]: Yes, it's unused now; removed.

File: apps/consent/app/consent/page.tsx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  consent_challenge: string
+}
+
+const submitForm = async (form: FormData) => {
+  "use server"
+  const consent_challenge = form.get("consent_challenge")
+  const submitValue = form.get("submit")
+  const remember = form.get("remember") === "1"
+  const grantScope = form.getAll("grant_scope").map((value) => value.toString())
+  // TODO add check from email code.
```
[the remark this answers is not in the document]
Comment (siddhart1o1) [reply apps/consent/app/consent/page.tsx]: nope I added that.

File: apps/consent/README.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# login-consent-provider
```
Comment (nicolasburtey) [inline apps/consent/README.md:1]: need better README to explain what consent is about.

`Consent`/Login is built on top of hydra to allow user to login and delegate access to their account.

File: apps/consent/app/consent/page.tsx [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+  const grantScope = form
+    .getAll("grant_scope")
+    .map((value) => value.toString());
+
+  if (!consent_challenge || typeof consent_challenge !== "string") {
+    console.error("INVALID PARAMS");
+    return;
+  }
+
+  if (submitValue === "Deny access") {
```
Comment (nicolasburtey) [inline apps/consent/app/consent/page.tsx:30]: `"Deny access"` should be a const

File: apps/consent/app/page.tsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+const Home = () => {
+  return (
+    <div>
+      <h1>Welcome to blink login!</h1>
+      <p>
+        This application will give you an idea of how you could implement the
+        login and consent endpoints yourself.
```
Comment (nicolasburtey) [inline apps/consent/app/page.tsx:9]: update text

File: apps/consent/app/oidc-cert.ts [hunk 11]
```diff
@@ -0,0 +1,89 @@
+// Copyright © 2023 Ory Corp
```
Comment (nicolasburtey) [inline apps/consent/app/oidc-cert.ts:1]: might not be needed?

**Pull request closed without merging**